l10n_pa_localization/model/res_partner.py

```diff
@@ -238,61 +230,45 @@ def _display_address(
             address_format = '%(company_name)s\n' + address_format
         return address_format % args
 
+    @api.onchange("hood_id")
     @api.multi
-    def onchange_hood_id(self, hood_id):
-        hood_id = self.env['res.country.state.district.township.hood'].browse(
-            hood_id)
-        value = {}
+    def _onchange_hood_id(self):
         for par in self:
-            value = {
-                'hood_id': hood_id.id,
-                'township_id': hood_id.township_id.id or par.township_id.id,
-                'district_id': hood_id.district_id.id or par.district_id.id,
-                'state_id': hood_id.state_id.id or par.state_id.id,
-                'country_id': hood_id.country_id.id or par. country_id.id
-            }
-        self.write(value)
-        return {'value': value}
+            par.township_id = par.hood_id.township_id or par.township_id
+            par.district_id = par.hood_id.district_id or par.district_id
```

Comment:
@hugho-ad 

If only you set the township_id, the change in that field will call the your onchange.

You not need assign all the values in this method

Can you try this?